src/Atmos/Model/filters.jl

```diff
@@ -51,3 +46,78 @@ function compute_filter_result!(
         filter_state.moisture.ρq_ice += aux.ref_state.ρq_ice
     end
 end
+
+struct AtmosSpecificFilterPerturbations{M} <: AbstractFilterTarget
+    atmos::M
+end
+
+vars_state_filtered(target::AtmosSpecificFilterPerturbations, FT) =
+    vars_state_filtered(target.atmos, FT)
+
+function compute_filter_argument!(
+    target::AtmosSpecificFilterPerturbations,
+    filter_state::Vars,
+    state::Vars,
+    aux::Vars,
+)
+
+    ρ_inv = 1 / state.ρ
+    filter_state.ρ = state.ρ - aux.ref_state.ρ
+
+    # remove reference state
+    filter_state.energy.e = (state.energy.ρe - aux.ref_state.ρe) * ρ_inv
+    if !(moisture_model(target.atmos) isa DryModel)
+        filter_state.moisture.q_tot =
+            (state.moisture.ρq_tot - aux.ref_state.ρq_tot) * ρ_inv
+    end
+    if (moisture_model(target.atmos) isa NonEquilMoist)
+        filter_state.moisture.q_liq =
+            (state.moisture.ρq_liq - aux.ref_state.ρq_liq) * ρ_inv
+        filter_state.moisture.q_ice =
+            (state.moisture.ρq_ice - aux.ref_state.ρq_ice) * ρ_inv
+    end
+
+    # turbconv vars:
+    prog_vars = prognostic_vars(turbconv_model(target.atmos))
+    map(prog_vars) do prog
+        fvar, fname = get_specific_state(filter_state, prog)
+        svar, sname = get_prog_state(state, prog)
+        val = getproperty(svar, sname) * ρ_inv
+        setproperty!(fvar, fname, val)
+    end
+end
+
+function compute_filter_result!(
+    target::AtmosSpecificFilterPerturbations,
+    state::Vars,
+    filter_state::Vars,
+    aux::Vars,
+)
+
+    state.ρ = filter_state.ρ + aux.ref_state.ρ
+    ρ = state.ρ # TODO: double check
```

Comment:
The density that we use to reconstruct the filtered state should not be filtered. Otherwise this is equivalent to the full filtering approach.

Reply:
What should this be, then?